### PR TITLE
Resolve auth issues

### PR DESCRIPTION
**Pull request by kylemh**
# Description of changes
- Use official example for [cookie auth in next](https://github.com/zeit/next.js/tree/canary/examples/with-cookie-auth)
- Resolve bug with `getInitialProps` not working (see new README + `withFonts` for details)
- Ensure that prod deploys get cookie properly
- Create utility method to provide displayName to any decorators

## Note
- Tests should fail because user creation is broken on staging BE momentarily

### TODO
1. Add unit tests for auth-utils
2. Add unit tests for cookie-utils

### CODE REVIEW

**Comment by kylemh** [comment]
🤖 **This is a bot** 🤖<br /><br />🎉 Deployed Storybook preview! 🎉<br /><br />Click the link at the bottom of this comment to see it live.

Built with commit 718ebeeacd2dfc12556d59e31ec2d97ac6d8828c

https://deploy-preview-425--operation-code-storybook.netlify.com

File: common/utils/auth-utils.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
+import Router from 'next/router';
+import isEmpty from 'lodash/isEmpty';
+import nextCookie from 'next-cookies';
+import { setAuthCookies, removeAuthCookies, isTokenValid } from './cookie-utils';
+
+export const login = async ({ token, user }) => {
```
Comment (thoughtbyte) [inline common/utils/auth-utils.js]: does this need to be async?

Comment (kylemh) [reply common/utils/auth-utils.js]: No it does not! Great catch.

**Comment by kylemh** [comment]
Skimping on tests here a bit, but that's okay. Most of this code comes from the official `with-cookie-auth` example on the Next repository.

**Pull request merged**